lib/modelValidator.js

```diff
@@ -387,6 +396,23 @@ function validateArray(name, value, field, models, allowBlankTargets, disallowEx
         }
         arrayType = fieldType;
     } else if(models && field.items && field.items.$ref) {
+        if (field.type && field.uniqueItems) {
+            var duplicatedObjects = []
+            for (var i = 0; i < value.length; i++) {
+                var j = i + 1
+                while(j < value.length && !doesArrayContainsValue(duplicatedObjects, value[j])) {
+                    if (_.isEqual(value[i], value[j])) {
+                        duplicatedObjects.push(value[i])
```

Comment:
Can we break out of both of these loops once we find a duplicate.  We only need to find one after all and then we don't need the duplicate check for the j loop and we can break the i loop too.

Reply:
yes, I'll change it

Reply:
fixed

Reply:
Looks good.